RecoLocalCalo/HcalRecAlgos/src/HFStripFilter.cc

```diff
@@ -0,0 +1,361 @@
+    }
+  
+  double ratio1 = eStrip > 0 ? energyIphi1/eStrip : 0;
+  double ratio2 = eStrip > 0 ? energyIphi2/eStrip : 0;
+    
+  if (verboseLevel_ >= 30) {
+    ss << "  iphi = " << stripIphiMax << "  iphi1 = " << iphi1 << "  iphi2 = " 
+       << iphi2 << std::endl
+       << "  Estrip = " << eStrip << "  EnergyIphi1 = " << energyIphi1
+       << "  Ratio = "	<< ratio1 << std::endl
+       << "                  " << "  EnergyIphi2 = " << energyIphi2
+       << "  Ratio = "	<< ratio2 << std::endl;
+  }
+  
+  // check if our wedge does not have substantial leak into adjacent wedges
+  if (ratio1 < wedgeCut_ && ratio2 < wedgeCut_) { // noise event with strips (d1 and/or d2)
+    
+    if (verboseLevel_ >= 30) {
+      ss << "  stripPass = false" << std::endl 
+	 << "  mark hits in strips now " << std::endl;
+    }
+   
+    // Figure out which rechits need to be tagged (d1.strip)
+    for (auto& it1 : d1strip) {
+      if (it1.id().ieta() < ietaMin || it1.id().ieta() > ietaMax) continue;
+      HFRecHitCollection::iterator hit = rec.find(it1.id());
+      if (hit != rec.end()) {
+	// tag a rechit with the anomalous hit flag
+	if (verboseLevel_ >= 30) {
+	  ss << " d1strip: marked hit = " << (*hit) << std::endl;
+	}
+	hit->setFlagField(1U, HcalPhase1FlagLabels::HFAnomalousHit);
+      }
+    }
+
+    // Figure out which rechits need to be tagged (d2.strip)
+    for (auto& it1 : d2strip) {
+      if (it1.id().ieta() < ietaMin || it1.id().ieta() > ietaMax) continue;
+      HFRecHitCollection::iterator hit = rec.find(it1.id());
+      if (hit != rec.end()) {
+	// tag a rechit with the anomalous hit flag
+	if (verboseLevel_ >= 30) {
+	  ss << " d2strip: marked hit = " << (*hit) << std::endl;
+	}
+	hit->setFlagField(1U, HcalPhase1FlagLabels::HFAnomalousHit);
+      }
+    }
+    if (verboseLevel_ >= 30) {
+      LogDebug("HFSFilter") << ss.str();
+    }
+  }
+  else {
+    if (verboseLevel_ >= 30) {
+      ss << "  stripPass = true" << std::endl;
+      LogDebug("HFSFilter") << ss.str();
+    }
+  } 
+}
+
+
+std::unique_ptr<HFStripFilter> HFStripFilter::parseParameterSet(
+    const edm::ParameterSet& ps)
+{
+    return std::make_unique<HFStripFilter>(
+        ps.getParameter<double>("stripThreshold"),
+        ps.getParameter<double>("maxThreshold"),
+        ps.getParameter<double>("timeMax"),
+        ps.getParameter<double>("maxStripTime"),
+        ps.getParameter<double>("wedgeCut"),
+        ps.getParameter<int>("seedHitIetaMax"),
+        ps.getParameter<int>("gap"),
+        ps.getParameter<int>("lstrips"),
+        ps.getUntrackedParameter<int>("verboseLevel")
+    );
+}
+
+edm::ParameterSetDescription HFStripFilter::fillDescription()
+{
+    edm::ParameterSetDescription desc;
+
+    desc.add<double>("stripThreshold", 40.0);
+    desc.add<double>("maxThreshold", 100.0);
+    desc.add<double>("timeMax", 6.0);
+    desc.add<double>("maxStripTime", 10.0);
+    desc.add<double>("wedgeCut", 0.05);
+    desc.add<int>("seedHitIetaMax", 35);
```

Comment:
When I asked whether this value could have been taken from somewhere else instead of hardcoding it, I had in mind somewhere in the geometry, rather than a configuration parameter here.

I expect that the eta/phi segmentation must be defined somewhere, and it would be preferable that you retrieve the details of that segmentation (e.g. ietaMax) from such a central location, so that if anything get changed centrally it can automatically propagate everywhere else.

Defining it with a configuration parameter inside a given method that needs to know that parameter can make sense if that parameter in this method can also take different values with respect to the central ones. Therefore, if this "35" refers to the fact that "the high-energy hits from these jets have ... Ieta mostly 29-34", having that "35" defined here does make sense. Otherwise, from the practical point of view, it is not that different than hardcoding it in the c++.

Reply:
Alexander, please check if something in the HcalTopology class (Geometry/​CaloTopology/​interface/​HcalTopology.h) corresponds to the magic number 35. If yes, we can forward a reference to the HcalTopology instance to your code. If not, I suggest leaving the current code as it is.